### PR TITLE
feat: ignore files in CodeCov + UTs reporting

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Ignore generated files in CodeCov
Add badge with reports into readme.md
Report UT results to CodeCov

## Related issues

*- No issue associated -*

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -10,3 +29,32 @@ component_management:
...
+        - apps/intellij-extension/**
+        - apps/vscode-extension/**
+    - component_id: otter-github-actions
+      name: Otter Github shared Actions
+      paths:
+        - tools/github-actions/**
+    - component_id: training
+      name: Training modules
+      paths:
+        - packages/@o3r/**
```
Comment (matthieu-crouzet) [inline codecov.yml:52]:
```suggestion
        - packages/@o3r-training/**
```

Comment (matthieu-crouzet) [inline codecov.yml:52]:
```suggestion
```

present without error just below

File: codecov.yml [hunk 2]
```diff
@@ -10,3 +29,32 @@ component_management:
...
       - type: project
         target: auto
+
+  individual_components:
+    - component_id: otter
+      name: Otter Modules
+      paths:
+        - packages/@o3r/**
+    - component_id: otter-tools
+      name: Otter tooling
```
Comment (matthieu-crouzet) [inline codecov.yml:39]:
```suggestion
      name: Otter developer tools
```

File: codecov.yml [hunk 3]
```diff
@@ -1,6 +1,25 @@
 codecov:
   require_ci_to_pass: false
 
+ignore:
+ # Package Manager files
```
Comment (fpaul-1A) [inline codecov.yml:5]:
```suggestion
  # Package Manager files
```

File: codecov.yml [hunk 4]
```diff
@@ -1,6 +1,25 @@
 codecov:
   require_ci_to_pass: false
 
+ignore:
```
Comment (fpaul-1A) [inline codecov.yml:4]: it's sad that there is no option to use `.gitignore` to ignore files out of the box

Comment (kpanot) [reply codecov.yml:4]: yes :'(

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**